File: 2. Add Two Numbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode r = l1, r1 = l1, r2 = l2;
        do{
            if(r2!=null){
                r1.val += r2.val;
                r2=r2.next;
            }
            if(r1.val / 10 != 0){
                if(r1.next == null)
                    r1.next = new ListNode(0, null);
                r1.next.val += r1.val / 10;
                r1.val %= 10;
            }
            
            
           
            if(r1.next==null && r2 != null){
            r1.next = new ListNode(0, null);
            }

            r1 = r1.next;
        }while(r1 != null);

        return r;
    }
}

File: 2. Add Two Numbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


int count = 0;
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    struct ListNode* l3;
    l3 = (struct ListNode *)malloc(sizeof(struct ListNode));
    
    if(l1 == NULL && l2 == NULL && count == 0)return NULL;
    if(l1 != NULL){
        count += l1->val;
        l1 = l1->next;
    }
    if(l2 != NULL){
        count += l2->val;
        l2 = l2->next;
    }
    
    l3->val = count%10;
    count/=10;
    
    l3->next = addTwoNumbers(l1, l2);
    return l3;
}
